### ISSUE TITLE
why doesn't Anaconda use rope library for refactoring?

### ISSUE THREAD

**Issue by divinites** [DamnWidget/anaconda#555]
I am curious, since Jedi library is not good at refactoring, so does anaconda. In the era of sublime text 2, we have sublimeRope, but now we have moved to ST3 for a long time, seems that all IDE-like python plug-in use Jedi, including YouCompleteMe SublimePythonIDE, etc.; no one uses rope anymore. Is there any compelling reason for this?

Moreover, Is there any better solution for python refactoring under ST3?

**Comment by DamnWidget** [comment]
I was regular contributor in SublimeRope, when the project was moved to SublimePythonIDE I helped with the transition and added linting support for it but at the end I started anaconda because difference of opinions about the course of SublimeRope/SublimePythonIDE.

Rope is a great tool it has many good features, one that I really miss a lot is it's ability to detect symbols in the buffer that were defined in a module in the `PYTHONPATH` but was not imported in the buffer, that was so cool to implement auto import tools, I implemented it in SublimeRope. 

Said that, rope is an extremely slow library that requires lots of caches and specific configurations that made it less flexible than Jedi. A part from that, Rope is a refactoring solution, not an auto-completion, fast code analysis one. I am not sure if things changed in the last years but one big difference between rope and jedi is that rope compiles/executes your code while jedi uses static analysis.

I am not aware of any good tool for refactoring in ST3 as I always refactor manually and actually the only Python projects that I work on are anaconda ones themselves.

Add rope to anaconda will be kind of trivial after the next [big refactor](https://github.com/DamnWidget/anaconda/projects) that is happening in the next/intermediate future of the package.
